### DISCUSSION TITLE
Dynamic Content and the modals manager

### DISCUSSION THREAD

**Discussion by mohammad-reza-rahbar** [mantinedev/mantine#6714] [Q&A]
I am using React Router [action](https://reactrouter.com/en/main/route/action) and [useFetcher](https://reactrouter.com/en/main/hooks/use-fetcher) with Mantine Modals manager. The problem is I want to change the state of the submit button inside the modal to loading state, but as mentioned [here](https://mantine.dev/x/modals/#dynamic-content-and-the-modals-manager), I should either use internal component state or the modal component instead of modals manager. My question is how can I use internal component state? Could you please give me an example?

**Comment by rtivital** [comment]
Use Modal component instead of modals manager

**Reply by mohammad-reza-rahbar** [reply]
I am aware of that. But the good thing about modals is that I can close it inside action. But for now, I don't know a clean way to close the Modal component after a successful action.

**Reply by rtivital** [reply]
Modal opened state is controlled by boolean value in `opened` prop, you can change it via `useState` hook

**Reply by bbuck** [reply]
Can you not handle dynamic content within a modal via tools like Zustand/redux and a custom Context modal component?
